app/mailboxes/support_mailbox.rb

```diff
@@ -58,12 +58,15 @@ def create_conversation
   end
 
   def find_or_create_contact
-    @contact = @inbox.contacts.find_by(email: processed_mail.from.first)
+    # changing to downcase to avoid case mismatch while finding
+    @contact = @inbox.contacts.find_by(email: processed_mail.from.first.downcase)
     if @contact.present?
       @contact_inbox = ContactInbox.find_by(inbox: @inbox, contact: @contact)
     else
       create_contact
     end
+  rescue StandardError => e
```

Comment:
We won't need this rescue as we intend for it to error out in unexpected cases. We capture it in sentry and fix it. Rescuing here, In a way, suppresses that intention. so let's remove this.

We could also add a `raise` after the logging statement in this rescue block where the original error was suppressed.

https://github.com/chatwoot/chatwoot/blob/a7ca55c0801561bbd0c4a9e21a775569914a7d6c/app/builders/contact_builder.rb#L58

Reply:
```suggestion
  rescue StandardError => e
     Rails.logger.info "#{e.message} in find/create_contact :: backtrace: #{e.backtrace}"
     raise e
   end
```
Shall I change like the above one?!

Reply:
@HariniKrishnan Let's  remove the rescue block from `app/mailboxes/support_mailbox.rb` all-together and place it in `chatwoot/app/builders/contact_builder.rb` instead with the `raise e` as you suggested. 

Since `create_contact` calls `ContactBuilder`, let's log the error at the root itself.

Reply:
sure will do that